### PR TITLE
Don't install pyboolector for ibex f4pga build 

### PR DESCRIPTION
**Pull request by kamilrakoczy**
pyboolector can't be build from wheel and some pip/python versions doesn't have binary release. We don't use this dependency.

This is workaround for currently not working ``[Large Designs Tests / Ibex (Symbiflow synthesis)]`` CI job:
```
08:12:27 |   × python setup.py egg_info did not run successfully.
08:12:27 |   │ exit code: 1
08:12:27 |   ╰─> [7 lines of output]
08:12:27 |       Traceback (most recent call last):
08:12:27 |         File "<string>", line 2, in <module>
08:12:27 |         File "<pip-setuptools-caller>", line 34, in <module>
08:12:27 |         File "/tmp/pip-install-fcq_197v/pyboolector_f3ecb0c085a04e96b5f23213dd3cda0b/setup.py", line 14, in <module>
08:12:27 |           with open(cmakelists_txt, "r") as f:
08:12:27 |                ^^^^^^^^^^^^^^^^^^^^^^^^^
08:12:27 |       FileNotFoundError: [Errno 2] No such file or directory: '/tmp/pip-install-fcq_197v/CMakeLists.txt'
08:12:27 |       [end of output]
08:12:27 |   
08:12:27 |   note: This error originates from a subprocess, and is likely not a problem with pip.
08:12:27 | error: metadata-generation-failed
08:12:27 | 
08:12:27 | × Encountered error while generating package metadata.
08:12:27 | ╰─> See above for output.
08:12:27 | 
08:12:27 | note: This is an issue with the package mentioned above, not pip.
08:12:27 | hint: See above for details.
08:12:27 | 
08:12:27 | 
08:12:27 | CondaEnvException: Pip failed
```

### CODE REVIEW

**Review by wsipak — APPROVED** [review]

**Review by mglb — CHANGES_REQUESTED** [review]
Please do this in https://github.com/antmicro/yosys-systemverilog/blob/f384153bcbda75157255d8f6d3469b1f4c30714c/uhdm-tests/ibex/Makefile.in to keep such things in one place.
Preferably using a regular patch, as this `sed` doesn't tell much and is unreliable (it will remove all `-r` from the requirements, in future possibly those that we want to keep).

**Comment by kamilrakoczy** [comment]
> Please do this in https://github.com/antmicro/yosys-systemverilog/blob/f384153bcbda75157255d8f6d3469b1f4c30714c/uhdm-tests/ibex/Makefile.in to keep such things in one place. Preferably using a regular patch, as this `sed` doesn't tell much and is unreliable (it will remove all `-r` from the requirements, in future possibly those that we want to keep).

I moved it to Makefile, but I needed to add this as another target.

The problem is that we are executing ``make env`` from included Makefile. In our Makefile.in we are only extending this command and third party ``make env`` is already trying to install ibex python dependencies. I don't think we can somehow add dependency to this third party ``env`` target.

**Comment by mglb** [comment]
`env` target doesn't read anything inside `ibex` repository, it happens in `

> > Please do this in https://github.com/antmicro/yosys-systemverilog/blob/f384153bcbda75157255d8f6d3469b1f4c30714c/uhdm-tests/ibex/Makefile.in to keep such things in one place. Preferably using a regular patch, as this `sed` doesn't tell much and is unreliable (it will remove all `-r` from the requirements, in future possibly those that we want to keep).
> 
> I moved it to Makefile, but I needed to add this as another target.
> 
> The problem is that we are executing `make env` from included Makefile. In our Makefile.in we are only extending this command and third party `make env` is already trying to install ibex python dependencies. I don't think we can somehow add dependency to this third party `env` target.

We can, but `env` is not what creates the environment - its dependency, `$(CONDA_ENV_PYTHON)`, does. The later conveniently depends on `${REQUIREMENTS_FILE}` (which makes sense, so it should never change), so we can add dependency to this file. However, we have to do it after conda.mk is included, as it changes `REQUIREMENTS_FILE` to an absolute path.

TL;DR: this seems to work:
```diff
diff --git a/uhdm-tests/ibex/Makefile.in b/uhdm-tests/ibex/Makefile.in
index 9cc582a..39c774d 100644
--- a/uhdm-tests/ibex/Makefile.in
+++ b/uhdm-tests/ibex/Makefile.in
@@ -66,17 +66,25 @@ ENVIRONMENT_FILE := ${curr_dir}/f4pga_environment.yml

 include ${curr_dir}/make-env/conda.mk

+# We need to apply the patch before Conda uses ${REQUIREMENTS_FILE}
+${REQUIREMENTS_FILE}: ${IBEX}/.requirementspatch
+
 # Install conda packages from f4pga/xc7 environment and set environment variables.
 env:: download-f4pga
-       ${IN_CONDA_ENV} conda env update --name ${CONDA_ENV_NAME} --file ${root_dir}/env/symbiflow/xc7_env/xc7_environment.yml
+       ${IN_CONDA_ENV} conda env update --name ${CONDA_ENV_NAME} --file ${root_dir}/env/symbiflow/xc7_env/xc7_environment.yml </dev/null | cat
        # Update edalize installed by Ibex to our own fork.
-       ${IN_CONDA_ENV} pip install git+https://github.com/antmicro/edalize@symbiflow-read_verilog_with_uhdm
+       #
+       ${IN_CONDA_ENV} pip install git+https://github.com/antmicro/edalize@symbiflow-read_verilog_with_uhdm </dev/null | cat
        ${IN_CONDA_ENV} conda env config vars set \
                F4PGA_INSTALL_DIR=${root_dir}/env/symbiflow \
-               F4PGA_SHARE_DIR=${root_dir}/env/symbiflow/share/f4pga
+               F4PGA_SHARE_DIR=${root_dir}/env/symbiflow/share/f4pga </dev/null | cat

 # Ibex build

+${IBEX}/.requirementspatch:
+       cd ${IBEX} && git apply ${curr_dir}/dont_install_riscv_dv_requirements.patch
+       touch $@
+
 ${IBEX}/.gitsymbiflowpatch: ${IBEX}/.gitpatch
        cd ${IBEX} && git apply ${curr_dir}/prim_generic_clock_gating.patch
        cd ${IBEX} && git apply ${curr_dir}/pins_artya7_xdc.patch && touch $@
```

**Comment by kamilrakoczy** [comment]
Oh, you are right. It looks much better now. Thank you.
